### PR TITLE
nodenext compatibility

### PR DESCRIPTION
**Pull request by Uzlopak**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: types/index.d.ts [hunk 1]
```diff
@@ -1,32 +1,36 @@
...
+    url?: string;
+    namespace?: string;
+  }) | {
+    client: Redis | Cluster;
+    namespace?: string;
+    closeClient?: boolean;
+  }
 
-/**
- * @deprecated Use `FastifyRedisPluginOptions` instead
```
Comment (Uzlopak) [inline types/index.d.ts:27]: i did not keep this deprecated type...
Breaking change?

Comment (climba03003) [reply types/index.d.ts:27]: Let keep it for now

Comment (Uzlopak) [reply types/index.d.ts:27]: done

**Review by Eomm — APPROVED** [review]

File: types/index.test-d.ts [hunk 2]
```diff
@@ -38,3 +38,5 @@ app.after(() => {
   expectType<Redis>(app.redis.one)
   expectType<Redis>(app.redis.two)
 })
+
+expectDeprecated({} as FastifyRedisPlugin)
```
Comment (Uzlopak) [inline types/index.test-d.ts:42]:
```suggestion
expectDeprecated({} as FastifyRedisPlugin)

```

**Review by climba03003 — APPROVED** [review]

**Pull request merged**